src/boot/efi/console.c

```diff
@@ -11,63 +11,107 @@
 
 #define EFI_SIMPLE_TEXT_INPUT_EX_GUID &(EFI_GUID) EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL_GUID
 
-EFI_STATUS console_key_read(UINT64 *key, BOOLEAN wait) {
+static inline void EventClosep(EFI_EVENT *event) {
+        if (!*event)
+                return;
+
+        uefi_call_wrapper(BS->CloseEvent, 1, *event);
+}
```

Comment:
ah, this is soo much prettier, i like!

Reply:
Yeah, I admit that once I wrote that little function I got a warm fuzzy feeling... :D